### PR TITLE
[13.0][MIG] hr_course

### PR DESCRIPTION
**Pull request by jarroyomorales**
Migration to v13 of `hr_course` 

@etobella @Saran440 @lfreeke @HviorForgeFlow

### CODE REVIEW

File: hr_course/models/hr_course.py [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+        readonly=True,
+        states={"in_validation": [("readonly", False)]},
+    )
+
+    @api.constrains("start_date", "end_date")
+    def _check_start_end_dates(self):
+        self.ensure_one()
+        if self.start_date and self.end_date and (self.start_date > self.end_date):
+            raise ValidationError(
+                _("The start date cannot" " be later than the end date.")
```
Comment (HviorForgeFlow) [inline hr_course/models/hr_course.py]:
```suggestion
                _("The start date cannot be later than the end date.")
```

File: hr_course/models/hr_course.py [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+    cost = fields.Monetary(string="Course Cost", required=True, tracking=True)
+    authorized_by = fields.Many2one(
+        string="Authorized by",
+        comodel_name="hr.employee",
+        required=True,
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+        tracking=True,
+    )
+    permanence = fields.Boolean(
```
Comment (HviorForgeFlow) [inline hr_course/models/hr_course.py:44]: help string would be useful here, to understand the configuration.

File: hr_course/models/hr_course.py [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+        string="Has Permanence",
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+        tracking=True,
+    )
+    permanence_time = fields.Char(
```
Comment (HviorForgeFlow) [inline hr_course/models/hr_course.py:52]: Not sure how to use this field ^^

**Review by HviorForgeFlow — APPROVED** [review]
Functional Review :+1:

Just a few optional comments

File: hr_course/models/hr_course.py (hunk 3 above)
Comment (jarroyomorales) [reply hr_course/models/hr_course.py:52]: This is an information field used to tell if making this course implies that employees have to stay in the company for at least a certain amount of time.

Comment (Saran440) [reply hr_course/models/hr_course.py:52]: @jarroyomorales Hi, this field mean employee take at least `permanence time` in course. right?

Comment (jarroyomorales) [reply hr_course/models/hr_course.py:52]: > This is an information field used to tell if making this course implies that employees have to stay in the company for at least a certain amount of time.

This was my idea, but maybe we need to rethink the usage of the field

Comment (etobella) [reply hr_course/models/hr_course.py:52]: I think the fields are legit, but, a help would be recommended

**Review by etobella — APPROVED** [review]

File: hr_course/models/hr_course.py (hunk 3 above)
Comment (jarroyomorales) [reply hr_course/models/hr_course.py:52]: done!

**Review by Saran440 — APPROVED** [review]
Functional Test :+1:

**Pull request merged**